global.d.ts

```diff
@@ -0,0 +1,36 @@
+interface PathProvider {
```

Comment:
Why declare this in a d.ts file? Normally I see stuff declared in .d.ts if we're defining TS shapes for JS files. In this case, since everything is TS to begin with, we may as well declare this interface in a normal .ts file.

I actually think that requiring users to import PathProvider is a feature not a bug - it's one less thing in global scope, plus it's easy to do auto-imports.

Reply:
Inlined the interface again. The imports seems to be okay. It is more in line with reducing global state, indeed.